h2o-py/h2o/explanation/_explain.py

```diff
@@ -1230,6 +1230,49 @@ def pd_multi_plot(
 
 def _center(col):
     col[:] = col - col[0]
+def _prepare_grouping_frames(frame, grouping_variable):
+    if (grouping_variable not in frame.names):
+        raise ValueError("Grouping variable '" + grouping_variable + "' is not present in frame!")
+    if (not frame[grouping_variable].isfactor()[0]):
+        raise ValueError("Grouping variable has to be categorical!")
+    categories = frame[grouping_variable].categories()
+    if len(categories) > 10:
+        raise ValueError("Grouping variable option is supported only for variables with 10 or fewer levels!")
+    frames = list()
+    for curr_category in categories:
+        key = "tmp_" + curr_category + str(random.randint(0,1000))
+        expr = "(tmp= " + key + "(rows " + frame.frame_id + " (==(cols " + frame.frame_id + " [" + str(frame.names.index(grouping_variable)) + "] ) '" + curr_category + "') ))"
```

Comment:
Oh no, what is this? 🙂  Is it going to be evaluated in python? Also I think much more readable alternative is to do string interpolation using `string.format()`

Reply:
nono, this is a rapids expression to be evaluated on java backend in this way: if there is a categorical variable for example color with values blue, black and white -> it will create 3 separate frames, one per each category: first with data only with blue rows, second with data with only black rows and third with data with only white rows. I don't mind changing to `string.format()`